#!/usr/bin/python3

# TODO!: fails on blue: 2022-05-03 13:02:16,168 INFO spawnerr: unknown error making dispatchers for 'syslog-ng': EACCES


import argparse, subprocess, sys, time

sys.path.append('/root/bin')
import kcore.docker_lib as D


def main():
    ap = argparse.ArgumentParser(description='syslogdock test')
    D.add_testing_args(ap)
    args = ap.parse_args()
    name, ip, cow, dv = D.launch_or_find_container(args)
    if not cow or not ip: sys.exit('cannot find container %s' % name)

    time.sleep(3)  # Give syslog a chance to start up.

    cookie = D.gen_random_cookie()
    subprocess.check_call(['/usr/bin/logger', '-n', ip, '-p', 'local1.info', '-P', '1514', cookie])

    time.sleep(2)  # Give syslog a chance to process.
    D.container_file_expect(cookie, name, '/var/log/queue')

    print('pass')
    

if __name__ == "__main__":
  main()

